src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -131,6 +162,128 @@ public class FhirCosmosSearchService : SearchService
             return searchResult;
         }
 
+        private async Task<List<Expression>> PerformChainedSearch(SearchOptions searchOptions, IReadOnlyList<ChainedExpression> chainedExpressions, CancellationToken cancellationToken)
+        {
+            if (!_cosmosConfig.EnableChainedSearch && _requestContextAccessor.FhirRequestContext?.RequestHeaders.TryGetValue(HeaderEnableChainedSearch, out StringValues _) != true)
```

Comment:
>&& [](start = 51, length = 2)

Shouldn't this be `||`
It's strange(at least for me ) to hide feature behind flag and header in same time.
I think what header should override server request in order to have granular control approach.